core/src/main/java/io/honeycomb/opentelemetry/android/Honeycomb.kt

```diff
@@ -133,18 +109,84 @@ class Honeycomb {
+                .build()
+        }
+
+        private fun buildSpanExporter(options: HoneycombOptions): SpanExporter {
+            val traceExporter = if (options.tracesProtocol == OtlpProtocol.GRPC) {
+                OtlpGrpcSpanExporter.builder()
+                    .setEndpoint(options.tracesEndpoint)
+                    .setTimeout(options.tracesTimeout.toJavaDuration())
+                    .setHeaders { options.tracesHeaders }
+                    .build()
+            } else {
+                OtlpHttpSpanExporter.builder()
+                    .setEndpoint(options.tracesEndpoint)
+                    .setTimeout(options.tracesTimeout.toJavaDuration())
+                    .setHeaders { options.tracesHeaders }
+                    .build()
+            }
 
             if (options.debug) {
-                otelRumBuilder.addSpanExporterCustomizer { OtlpJsonLoggingSpanExporter.create() }
-                otelRumBuilder.addMeterProviderCustomizer{ builder, _ ->
-                    builder.registerMetricReader(
-                        PeriodicMetricReader.builder(OtlpJsonLoggingMetricExporter.create()).build()
-                    )
-                }
+                return SpanExporter.composite(traceExporter, OtlpJsonLoggingSpanExporter.create())
+            }
+            return traceExporter;
+        }
+
+        private fun buildMetricsExporter(options: HoneycombOptions): MetricExporter {
+            val metricsExporter = if (options.metricsProtocol == OtlpProtocol.GRPC) {
+                OtlpGrpcMetricExporter.builder()
+                    .setEndpoint(options.metricsEndpoint)
+                    .setTimeout(options.metricsTimeout.toJavaDuration())
+                    .setHeaders { options.metricsHeaders }
+                    .build()
+            } else {
+                OtlpHttpMetricExporter.builder()
+                    .setEndpoint(options.metricsEndpoint)
+                    .setTimeout(options.metricsTimeout.toJavaDuration())
+                    .setHeaders { options.metricsHeaders }
+                    .build()
             }
 
-            return otelRumBuilder.build()
+            if (options.debug) {
+                return CompositeMetricExporter(metricsExporter, OtlpJsonLoggingMetricExporter.create())
+            }
+            return metricsExporter;
         }
     }
-}
 
+    private class CompositeMetricExporter(vararg val exporters: MetricExporter): MetricExporter {
+        override fun getAggregationTemporality(instrumentType: InstrumentType): AggregationTemporality {
+            return try {
+                exporters.first().getAggregationTemporality(instrumentType);
+            } catch (e: NoSuchElementException) {
+                // doesn't really matter
+                AggregationTemporality.DELTA;
+            }
+        }
+
+        override fun export(metrics: MutableCollection<MetricData>): CompletableResultCode {
+            val codes = ArrayList<CompletableResultCode>()
```

Comment:
This is good. I am curious whether it would be more idiomatic in Kotlin to write:

```kotlin
            return CompletableResultCode.ofAll(exporters.map { it.export(metrics) });
```

Totally up to you, assuming that even compiles.

Reply:
I considered this but it felt _weird_ to use `map` for the side effect of the `export()` call.